src/components/TodoModal/useTodoModal.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+export const useTodoModal = (todoId: number | null) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      if (todoId !== null) {
+        const userData = await getUser(todoId);
+
+        setUser(userData);
+        setIsLoading(false);
```

Comment:
here you also need error handling ;) remember to always handle failures in your asynchronous code to ensure a better user experience and prevent uncaught errors